lib/db.js

```diff
@@ -59,10 +59,18 @@ module.exports = (
       const redisConfig = {
         host: config.redis.host,
         port: config.redis.port,
-        prefix: config.redis.sessionsKeyPrefix
+        prefix: config.redis.sessionsKeyPrefix,
+        // Prefer redis to fail fast than wait indefinitely for reconnection
+        enable_offline_queue: false
       }
       log.info({op: 'db.redis.enabled', config: redisConfig})
       this.redis = redis.createClient(redisConfig)
+      // If the `error` event is not handled, redis errors are fatal
+      this.redis.on('error', err => log.error({
```

Comment:
@philbooth What are your thoughts on having the `isRedisOk` be global and triggered from connection [events](https://github.com/NodeRedis/node_redis#connection-and-other-events)?

Reply:
That would lead to race conditions I think, albeit rare ones. In the time between checking `isRedisOk` and setting `lastAccessTime`, the state may flip from `true` to `false` and we'd return a stale timestamp. Better to keep it local and then just try the get to see if it works.